### PR TITLE
fix(alert): prevent page styles from overriding component styles

### LINKED ISSUE #842: [bug]: alert heading is not inline with icon
Alert heading is not inline with icon

see https://deploy-preview-821--red-hat-design-system.netlify.app/elements/call-to-action/#navigation-vs.-action

_Originally posted by @nikkimk in https://github.com/RedHat-UX/red-hat-design-system/pull/821#discussion_r1146416838_

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. sprinkle `!important` on to slotted styles

Closes #842

### CODE REVIEW

File: elements/rh-alert/rh-alert.css [hunk 1]
```diff
@@ -119,34 +119,33 @@ header ::slotted(:is(h1,h2,h3,h4,h5,h6)) {
...
-  margin-bottom: var(--rh-space-lg, 16px);
+  margin-block: var(--rh-space-md, 8px) var(--rh-space-lg, 16px) !important;
 }
 
 footer.hasActions {
   margin-top: var(--rh-space-lg, 16px);
 }
 
 footer ::slotted([slot="actions"]) {
-  margin-right: var(--actions-item--MarginRight, var(--rh-alert__actions--Gap, 24px)) !important;
```
Comment (heyMP) [inline elements/rh-alert/rh-alert.css:131]: 👍

**Comment by heyMP** [comment]
Can we make a hard and fast rule saying that all `::slotted()` rules need to include `!important`?

**Comment by bennypowers** [comment]
> ~Can~ Should we make a hard and fast rule saying that all `::slotted()` rules need to include `!important`?

"hard and fast" is maybe too far, but yes, in cases like this where we know what the style should be in all cases, we're allowed to - and are supposed to - clobber light styles

File: elements/rh-alert/rh-alert.css (hunk 1 above)
Comment (bennypowers) [reply elements/rh-alert/rh-alert.css:131]: yeah i saw that and I was like 

"ay yay yay yay yay!"

**Review by brianferry — CHANGES_REQUESTED** [review]
Resolve to merge

**Review by brianferry — APPROVED** [review]
Looks good, tested in docs/ia-cta branch and looks good to me

**Pull request merged**